app/services/approval_service_api.rb

```diff
@@ -0,0 +1,25 @@
+require 'approval_api_client'
+
+class ApprovalServiceApi
+  attr_reader :params
+  attr_reader :api_instance
+  attr_reader :request
+
+  def initialize(options)
+    @params = options[:params]
+    @request = options[:request]
+    ApprovalAPIClient.configure do |config|
+      config.host     = ENV['APPROVAL_SERVICE_URL']
+      config.scheme   = URI.parse(ENV['APPROVAL_SERVICE_URL']).try(:scheme)
+    end
+    @api_instance = ApprovalAPIClient::UsersApi.new
+    set_identity_header
+  end
+
+  def set_identity_header
+    x_rh = {
```

Comment:
It would be cool to wrap all this header work into this library, https://github.com/ManageIQ/service_portal-api/pull/12/files#diff-185bb06aaf70b927dd20da06f6fd70c6 but it isn't merged yet.